### PR TITLE
Automatic fire for mobs

### PR DESCRIPTION
**Pull request by Jacquerel**
## About The Pull Request

Written for #79473 but that PR is already big enough
This adds a component you can slap onto mobs with ranged attacks which lets them fire repeatedly with the mouse held down.
I applied it to a couple of existing mobs, to be honest we don't currently have any which _need_ this but I will add it to the Ranged Holoparasite (which really does need it) depending on which PR gets merged first.

## Why It's Good For The Game

This allows mobs to utilise a high fire rate without giving you hand strain.

## Changelog

:cl:
qol: Adds the capability for some player-controlled mobs with ranged attacks to repeatedly fire their natural weapons by holding down the mouse button.
/:cl:

### CODE REVIEW

**Comment by necromanceranne** [comment]
I'm pretty sure the Nanotrasen swat mobs use WT-550's don't they? They could use it.

**Comment by Jacquerel** [comment]
> I'm pretty sure the Nanotrasen swat mobs use WT-550's don't they? They could use it.

Currently that shoots three round bursts with a slow fire rate, presumably because simple mobs weren't actually able to shoot that fast? Making them shoot faster just makes them absurdly deadly though.
I'm not sure what these mobs are for in the first place but I don't know if I want to mess with it.

**Comment by necromanceranne** [comment]
The WT-550 used to be a three round burst. It is just out of date to the new version of the WT-550. 

They're used in a few away missions. One of them is current.

File: code/datums/components/ranged_mob_full_auto.dm [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+	if (LAZYACCESS(modifiers, SHIFT_CLICK))
+		return
+	if (LAZYACCESS(modifiers, CTRL_CLICK))
+		return
+	if (LAZYACCESS(modifiers, MIDDLE_CLICK))
+		return
+	if (LAZYACCESS(modifiers, RIGHT_CLICK))
+		return
+	if (LAZYACCESS(modifiers, ALT_CLICK))
+		return
```
Comment (san7890) [inline code/datums/components/ranged_mob_full_auto.dm:111]: we really need to have a `LAZYACCESS()` proc-like helper that can check for a match in a list of stuff this is insane

**Review by san7890 — COMMENTED** [review]
you don' tneed to do this now but this is pretty bad looking

**Pull request merged**